state/upgrades.go

```diff
@@ -65,6 +65,40 @@ func replaceBsonDField(d bson.D, name string, value interface{}) error {
 	return errors.NotFoundf("field %q", name)
 }
 
+// RenameAddModelPermission renames any permissions called addmodel to add-model.
+func RenameAddModelPermission(st *State) error {
+	coll, closer := st.getRawCollection(permissionsC)
+	defer closer()
+	upgradesLogger.Infof("migrating addmodel permission")
+
+	iter := coll.Find(nil).Iter()
+	defer iter.Close()
+	var ops []txn.Op
+	var doc bson.M
+	for iter.Next(&doc) {
+		fieldVal, ok := doc["access"]
+		if !ok || fieldVal != "addmodel" {
```

Comment:
You could avoid this and make the set of documents shorter by querying for permissions which use `addmodel`.